refactor(srgssr-middleware): move srgssr specific options declaration

jboix opened the pull request:
## Description

Considering we generate two builds for the library it makes sense to perform the declaration of SRG SSR specific options in the middleware itself. This approach prevents the `core` build from exposing options specific to the SRG SSR business logic and ensures these options are only present when necessary.

## Changes made

- Moved the declaration of SRG SSR specific options from `src/pillarbox.js` to `src/middleware/srgssr.js`.

amtins commented:
@jboix While I agree that `srgOptions` should be moved as it only concerns SRG SSR, I believe that the `trackers` option should remain in `pillarbox`.

The original intention is to provide a placeholder for accessing the various trackers directly from the player. Even if the use of this placeholder is optional, its use case remains valid.

Would it be possible to change the commit header from `refactor(middleware)` to `refactor(srgssr-middleware)`?

jboix commented:
As requested in: https://github.com/SRGSSR/pillarbox-web/pull/209#issuecomment-1953795954

- Moved `trackers` back to `pillarbox.js`. I 've added some comment to this one so that the intention is explicit.
- Changed the commit message.

amtins reviewed (approved):
LGTM!
